tests/app/letters/test_letter_utils.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+from flask import current_app
+
+from app.letters.utils import get_bucket_prefix_for_notification, is_precompiled_letter
+
+
+def test_get_bucket_prefix_for_notification_valid_notification(sample_notification):
+
+    bucket_prefix = get_bucket_prefix_for_notification(sample_notification)
+
+    assert bucket_prefix == '{folder}/NOTIFY.{reference}'.format(
+        folder=sample_notification.created_at.date(),
+        reference=sample_notification.reference
+    ).upper()
+
+
+def test_get_bucket_prefix_for_notification_invalid_notification():
+    with pytest.raises(AttributeError):
+        get_bucket_prefix_for_notification(None)
+
+
+def test_is_precompiled_letter_false(sample_letter_template):
```

Comment:
Might also want to check conditions where it is hidden but doesn't have the correct template name and where it does have the correct template name but is not hidden